### ISSUE TITLE
Linux Build Issues

### ISSUE THREAD

**Issue by belak** [cocos2d/cocos2d-x#2215]
Here's the output from COCOS2DX_USEAPT=false ./make-all-linux-project.sh                                    

```
make -C external/chipmunk/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libchipmunk.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libbox2d.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libcocosdenshion.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libextension.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make -C cocos2dx/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libcocos2d.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
make -C scripting/lua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/liblua.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
make -C samples/Cpp/HelloCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
rm -rf obj/debug
rm -f bin/debug/HelloCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
make -C samples/Cpp/TestCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
rm -rf obj/debug
rm -f bin/debug/TestCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
make -C samples/Cpp/SimpleGame/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
rm -rf obj/debug
rm -f bin/debug/SimpleGame core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
make -C samples/Lua/HelloLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
rm -rf obj/debug
rm -f bin/debug/HelloLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
make -C samples/Lua/TestLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
rm -rf obj/debug
rm -f bin/debug/TestLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
make -C external/chipmunk/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/libchipmunk.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/libbox2d.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libcocosdenshion.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libextension.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make -C cocos2dx/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libcocos2d.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
make -C scripting/lua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/liblua.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
make -C samples/Cpp/HelloCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
rm -rf obj/release
rm -f bin/release/HelloCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
make -C samples/Cpp/TestCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
rm -rf obj/release
rm -f bin/release/TestCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
make -C samples/Cpp/SimpleGame/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
rm -rf obj/release
rm -f bin/release/SimpleGame core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
make -C samples/Lua/HelloLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
rm -rf obj/release
rm -f bin/release/HelloLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
make -C samples/Lua/TestLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
rm -rf obj/release
rm -f bin/release/TestLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
make -C external/chipmunk/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
 CC obj/debug/cpBody.o
 CC obj/debug/chipmunk.o
 CC obj/debug/cpSpace.o
 CC obj/debug/cpSpatialIndex.o
 CC obj/debug/cpArbiter.o
 CC obj/debug/cpCollision.o
 CC obj/debug/cpSpaceComponent.o
 CC obj/debug/cpSweep1D.o
 CC obj/debug/cpArray.o
 CC obj/debug/cpHashSet.o
 CC obj/debug/cpSpaceHash.o
 CC obj/debug/cpVect.o
 CC obj/debug/cpBB.o
 CC obj/debug/cpPolyShape.o
 CC obj/debug/cpSpaceQuery.o
 CC obj/debug/cpBBTree.o
 CC obj/debug/cpShape.o
 CC obj/debug/cpSpaceStep.o
 CC obj/debug/constraints/cpConstraint.o
 CC obj/debug/constraints/cpPivotJoint.o
 CC obj/debug/constraints/cpDampedRotarySpring.o
 CC obj/debug/constraints/cpRatchetJoint.o
 CC obj/debug/constraints/cpDampedSpring.o
 CC obj/debug/constraints/cpRotaryLimitJoint.o
 CC obj/debug/constraints/cpGearJoint.o
 CC obj/debug/constraints/cpSimpleMotor.o
 CC obj/debug/constraints/cpGrooveJoint.o
 CC obj/debug/constraints/cpSlideJoint.o
 CC obj/debug/constraints/cpPinJoint.o
 AR /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libchipmunk.a
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
 CXX obj/debug/Collision/Shapes/b2CircleShape.o
 CXX obj/debug/Collision/Shapes/b2EdgeShape.o
 CXX obj/debug/Collision/Shapes/b2PolygonShape.o
 CXX obj/debug/Collision/Shapes/b2ChainShape.o
 CXX obj/debug/Collision/b2BroadPhase.o
 CXX obj/debug/Collision/b2CollideCircle.o
 CXX obj/debug/Collision/b2CollideEdge.o
 CXX obj/debug/Collision/b2CollidePolygon.o
 CXX obj/debug/Collision/b2Collision.o
 CXX obj/debug/Collision/b2Distance.o
 CXX obj/debug/Collision/b2DynamicTree.o
 CXX obj/debug/Collision/b2TimeOfImpact.o
 CXX obj/debug/Common/b2BlockAllocator.o
 CXX obj/debug/Common/b2Draw.o
 CXX obj/debug/Common/b2Math.o
 CXX obj/debug/Common/b2Settings.o
 CXX obj/debug/Common/b2StackAllocator.o
 CXX obj/debug/Common/b2Timer.o
 CXX obj/debug/Dynamics/Contacts/b2ChainAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2ChainAndPolygonContact.o
 CXX obj/debug/Dynamics/Contacts/b2CircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2Contact.o
 CXX obj/debug/Dynamics/Contacts/b2ContactSolver.o
 CXX obj/debug/Dynamics/Contacts/b2EdgeAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2EdgeAndPolygonContact.o
 CXX obj/debug/Dynamics/Contacts/b2PolygonAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2PolygonContact.o
 CXX obj/debug/Dynamics/Joints/b2DistanceJoint.o
 CXX obj/debug/Dynamics/Joints/b2FrictionJoint.o
 CXX obj/debug/Dynamics/Joints/b2GearJoint.o
 CXX obj/debug/Dynamics/Joints/b2Joint.o
 CXX obj/debug/Dynamics/Joints/b2MouseJoint.o
 CXX obj/debug/Dynamics/Joints/b2PrismaticJoint.o
 CXX obj/debug/Dynamics/Joints/b2PulleyJoint.o
 CXX obj/debug/Dynamics/Joints/b2RevoluteJoint.o
 CXX obj/debug/Dynamics/Joints/b2RopeJoint.o
 CXX obj/debug/Dynamics/Joints/b2WeldJoint.o
 CXX obj/debug/Dynamics/Joints/b2WheelJoint.o
 CXX obj/debug/Dynamics/b2Body.o
 CXX obj/debug/Dynamics/b2ContactManager.o
 CXX obj/debug/Dynamics/b2Fixture.o
 CXX obj/debug/Dynamics/b2Island.o
 CXX obj/debug/Dynamics/b2World.o
 CXX obj/debug/Dynamics/b2WorldCallbacks.o
 CXX obj/debug/Rope/b2Rope.o
 AR /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libbox2d.a
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
 CXX obj/debug/linux/FmodAudioPlayer.o
 CXX obj/debug/linux/SimpleAudioEngine.o
 LINK ../../cocos2dx/lib/linux/debug/libcocosdenshion.so
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
 CXX obj/debug/CCBReader/CCBFileLoader.o
 CXX obj/debug/CCBReader/CCMenuItemImageLoader.o
 CXX obj/debug/CCBReader/CCBReader.o
 CXX obj/debug/CCBReader/CCMenuItemLoader.o
 CXX obj/debug/CCBReader/CCNodeLoader.o
 CXX obj/debug/CCBReader/CCControlButtonLoader.o
 CXX obj/debug/CCBReader/CCControlLoader.o
 CXX obj/debug/CCBReader/CCNodeLoaderLibrary.o
 CXX obj/debug/CCBReader/CCLabelBMFontLoader.o
 CXX obj/debug/CCBReader/CCParticleSystemQuadLoader.o
 CXX obj/debug/CCBReader/CCLabelTTFLoader.o
 CXX obj/debug/CCBReader/CCScale9SpriteLoader.o
 CXX obj/debug/CCBReader/CCLayerColorLoader.o
../CCBReader/CCNodeLoader.cpp: In member function ‘virtual float* cocos2d::extension::CCNodeLoader::parsePropTypeFloatXY(cocos2d::CCNode*, cocos2d::CCNode*, cocos2d::extension::CCBReader*)’:
../CCBReader/CCNodeLoader.cpp:454:9: error: unused variable ‘type’ [-Werror=unused-variable]
 CXX obj/debug/CCBReader/CCScrollViewLoader.o
cc1plus: all warnings being treated as errors
make[1]: *** [obj/debug/CCBReader/CCNodeLoader.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make: *** [all] Error 2
```

If I modify cocos2dx/proj.linux/cocos2dx.mk and remove -Werror, this is what happens:

```
make -C external/chipmunk/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libchipmunk.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libbox2d.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libcocosdenshion.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libextension.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make -C cocos2dx/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
rm -rf obj/debug
rm -f ../../cocos2dx/lib/linux/debug/libcocos2d.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
make -C scripting/lua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
rm -rf obj/debug
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/liblua.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
make -C samples/Cpp/HelloCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
rm -rf obj/debug
rm -f bin/debug/HelloCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
make -C samples/Cpp/TestCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
rm -rf obj/debug
rm -f bin/debug/TestCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
make -C samples/Cpp/SimpleGame/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
rm -rf obj/debug
rm -f bin/debug/SimpleGame core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
make -C samples/Lua/HelloLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
rm -rf obj/debug
rm -f bin/debug/HelloLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
make -C samples/Lua/TestLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
rm -rf obj/debug
rm -f bin/debug/TestLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
make -C external/chipmunk/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/libchipmunk.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/libbox2d.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libcocosdenshion.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libextension.a core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make -C cocos2dx/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
rm -rf obj/release
rm -f ../../cocos2dx/lib/linux/release/libcocos2d.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/cocos2dx/proj.linux'
make -C scripting/lua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
rm -rf obj/release
rm -f /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/release/liblua.so core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/scripting/lua/proj.linux'
make -C samples/Cpp/HelloCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
rm -rf obj/release
rm -f bin/release/HelloCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/HelloCpp/proj.linux'
make -C samples/Cpp/TestCpp/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
rm -rf obj/release
rm -f bin/release/TestCpp core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/TestCpp/proj.linux'
make -C samples/Cpp/SimpleGame/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
rm -rf obj/release
rm -f bin/release/SimpleGame core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Cpp/SimpleGame/proj.linux'
make -C samples/Lua/HelloLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
rm -rf obj/release
rm -f bin/release/HelloLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/HelloLua/proj.linux'
make -C samples/Lua/TestLua/proj.linux clean
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
rm -rf obj/release
rm -f bin/release/TestLua core
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/samples/Lua/TestLua/proj.linux'
make -C external/chipmunk/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
 CC obj/debug/chipmunk.o
 CC obj/debug/cpBody.o
 CC obj/debug/cpSpace.o
 CC obj/debug/cpSpatialIndex.o
 CC obj/debug/cpArbiter.o
 CC obj/debug/cpCollision.o
 CC obj/debug/cpSpaceComponent.o
 CC obj/debug/cpSweep1D.o
 CC obj/debug/cpArray.o
 CC obj/debug/cpHashSet.o
 CC obj/debug/cpSpaceHash.o
 CC obj/debug/cpVect.o
 CC obj/debug/cpBB.o
 CC obj/debug/cpPolyShape.o
 CC obj/debug/cpSpaceQuery.o
 CC obj/debug/cpBBTree.o
 CC obj/debug/cpShape.o
 CC obj/debug/cpSpaceStep.o
 CC obj/debug/constraints/cpConstraint.o
 CC obj/debug/constraints/cpPivotJoint.o
 CC obj/debug/constraints/cpDampedRotarySpring.o
 CC obj/debug/constraints/cpRatchetJoint.o
 CC obj/debug/constraints/cpDampedSpring.o
 CC obj/debug/constraints/cpRotaryLimitJoint.o
 CC obj/debug/constraints/cpGearJoint.o
 CC obj/debug/constraints/cpSimpleMotor.o
 CC obj/debug/constraints/cpGrooveJoint.o
 CC obj/debug/constraints/cpSlideJoint.o
 CC obj/debug/constraints/cpPinJoint.o
 AR /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libchipmunk.a
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/chipmunk/proj.linux'
make -C external/Box2D/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
 CXX obj/debug/Collision/Shapes/b2CircleShape.o
 CXX obj/debug/Collision/Shapes/b2ChainShape.o
 CXX obj/debug/Collision/Shapes/b2EdgeShape.o
 CXX obj/debug/Collision/Shapes/b2PolygonShape.o
 CXX obj/debug/Collision/b2BroadPhase.o
 CXX obj/debug/Collision/b2CollideCircle.o
 CXX obj/debug/Collision/b2CollideEdge.o
 CXX obj/debug/Collision/b2CollidePolygon.o
 CXX obj/debug/Collision/b2Collision.o
 CXX obj/debug/Collision/b2Distance.o
 CXX obj/debug/Collision/b2DynamicTree.o
 CXX obj/debug/Collision/b2TimeOfImpact.o
 CXX obj/debug/Common/b2BlockAllocator.o
 CXX obj/debug/Common/b2Draw.o
 CXX obj/debug/Common/b2Math.o
 CXX obj/debug/Common/b2Settings.o
 CXX obj/debug/Common/b2StackAllocator.o
 CXX obj/debug/Common/b2Timer.o
 CXX obj/debug/Dynamics/Contacts/b2ChainAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2ChainAndPolygonContact.o
 CXX obj/debug/Dynamics/Contacts/b2CircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2Contact.o
 CXX obj/debug/Dynamics/Contacts/b2ContactSolver.o
 CXX obj/debug/Dynamics/Contacts/b2EdgeAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2EdgeAndPolygonContact.o
 CXX obj/debug/Dynamics/Contacts/b2PolygonAndCircleContact.o
 CXX obj/debug/Dynamics/Contacts/b2PolygonContact.o
 CXX obj/debug/Dynamics/Joints/b2DistanceJoint.o
 CXX obj/debug/Dynamics/Joints/b2FrictionJoint.o
 CXX obj/debug/Dynamics/Joints/b2GearJoint.o
 CXX obj/debug/Dynamics/Joints/b2Joint.o
 CXX obj/debug/Dynamics/Joints/b2MouseJoint.o
 CXX obj/debug/Dynamics/Joints/b2PrismaticJoint.o
 CXX obj/debug/Dynamics/Joints/b2PulleyJoint.o
 CXX obj/debug/Dynamics/Joints/b2RevoluteJoint.o
 CXX obj/debug/Dynamics/Joints/b2RopeJoint.o
 CXX obj/debug/Dynamics/Joints/b2WeldJoint.o
 CXX obj/debug/Dynamics/Joints/b2WheelJoint.o
 CXX obj/debug/Dynamics/b2Body.o
 CXX obj/debug/Dynamics/b2ContactManager.o
 CXX obj/debug/Dynamics/b2Fixture.o
 CXX obj/debug/Dynamics/b2Island.o
 CXX obj/debug/Dynamics/b2World.o
 CXX obj/debug/Dynamics/b2WorldCallbacks.o
 CXX obj/debug/Rope/b2Rope.o
 AR /home/belak/.runtime/cocos2d-x/cocos2dx/lib/linux/debug/libbox2d.a
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/external/Box2D/proj.linux'
make -C CocosDenshion/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
 CXX obj/debug/linux/FmodAudioPlayer.o
 CXX obj/debug/linux/SimpleAudioEngine.o
 LINK ../../cocos2dx/lib/linux/debug/libcocosdenshion.so
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/CocosDenshion/proj.linux'
make -C extensions/proj.linux all
make[1]: Entering directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
 CXX obj/debug/CCBReader/CCMenuItemImageLoader.o
 CXX obj/debug/CCBReader/CCBFileLoader.o
 CXX obj/debug/CCBReader/CCBReader.o
 CXX obj/debug/CCBReader/CCMenuItemLoader.o
 CXX obj/debug/CCBReader/CCControlButtonLoader.o
 CXX obj/debug/CCBReader/CCNodeLoader.o
 CXX obj/debug/CCBReader/CCControlLoader.o
 CXX obj/debug/CCBReader/CCNodeLoaderLibrary.o
 CXX obj/debug/CCBReader/CCLabelBMFontLoader.o
 CXX obj/debug/CCBReader/CCParticleSystemQuadLoader.o
../CCBReader/CCNodeLoader.cpp: In member function ‘virtual float* cocos2d::extension::CCNodeLoader::parsePropTypeFloatXY(cocos2d::CCNode*, cocos2d::CCNode*, cocos2d::extension::CCBReader*)’:
../CCBReader/CCNodeLoader.cpp:454:9: warning: unused variable ‘type’ [-Wunused-variable]
 CXX obj/debug/CCBReader/CCLabelTTFLoader.o
 CXX obj/debug/CCBReader/CCScale9SpriteLoader.o
 CXX obj/debug/CCBReader/CCLayerColorLoader.o
 CXX obj/debug/CCBReader/CCScrollViewLoader.o
 CXX obj/debug/CCBReader/CCLayerGradientLoader.o
 CXX obj/debug/CCBReader/CCSpriteLoader.o
 CXX obj/debug/CCBReader/CCLayerLoader.o
 CXX obj/debug/CCBReader/CCBAnimationManager.o
../CCBReader/CCBAnimationManager.cpp:9:31: fatal error: SimpleAudioEngine.h: No such file or directory
compilation terminated.
make[1]: *** [obj/debug/CCBReader/CCBAnimationManager.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/home/belak/.runtime/cocos2d-x/extensions/proj.linux'
make: *** [all] Error 2
```

Is it something I'm doing wrong, or is there an issue with the current head?

**Comment by belak** [comment]
Fixed in 441321e1a18e4286df7af37a618123ec2488f84c

**Comment by dumganhar** [comment]
@belak 
Thanks for your feedback. :)
